universal-application-tool-0.0.1/app/controllers/admin/AdminApplicationController.java

```diff
@@ -23,22 +34,48 @@
   private final ApplicationRepository applicationRepository;
   private final ProgramApplicationListView applicationListView;
   private final ProgramApplicationView applicationView;
+  private final ExporterFactory exporterFactory;
+  private final Clock clock;
 
   @Inject
   public AdminApplicationController(
       ProgramService service,
       ProgramApplicationListView applicationListView,
       ProgramApplicationView applicationView,
-      ApplicationRepository applicationRepository) {
+      ApplicationRepository applicationRepository,
+      ExporterFactory exporterFactory,
+      Clock clock) {
     this.service = checkNotNull(service);
     this.applicationListView = checkNotNull(applicationListView);
     this.applicationView = checkNotNull(applicationView);
     this.applicationRepository = checkNotNull(applicationRepository);
+    this.exporterFactory = checkNotNull(exporterFactory);
+    this.clock = clock;
   }
 
   @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
   public Result downloadAll(long programId) {
-    throw new UnsupportedOperationException("Not yet implemented.");
+    try {
+      ImmutableList<Application> applications = service.getProgramApplications(programId);
+      ProgramDefinition program = service.getProgramDefinition(programId);
+      CsvExporter csvExporter = exporterFactory.csvExporter(program.toProgram());
+
+      OutputStream inMemoryBytes = new ByteArrayOutputStream();
+      Writer writer = new OutputStreamWriter(inMemoryBytes, StandardCharsets.UTF_8);
+      for (Application application : applications) {
+        csvExporter.export(application.getApplicantData(), writer);
+      }
+      writer.close();
+      String filename = String.format("%s-%s.csv", program.name(), clock.instant().toString());
```

Comment:
Suggestion (blocking): please move most of this logic into `ApplicantService` or consider creating a `CsvService`, see [separation of concerns](https://github.com/seattle-uat/civiform/wiki/Development-standards#separation-of-concerns) guidelines.

Reply:
Moved to `programService` since "get a csv" is more of a program operation.

Reply:
Looks like that introduces a circular dependency between the exporter and program service. How about creating a new `CsvService`or `ExportService`?